### PR TITLE
fix onnximport's dt

### PR DESCRIPTION
**Pull request by HeJunchao100813**

### CODE REVIEW

File: src/Nncase.Importer/Onnx/Reduce.cs [hunk 1]
```diff
@@ -11,12 +11,23 @@ namespace Nncase.Importer
...
     {
-        private Expr VisitReduce(in NodeProto op, ReduceOp reduceOp, float initValue)
+        private Expr VisitReduce(in NodeProto op, ReduceOp reduceOp, Expr initValue)
         {
-            return ReduceCore(op, reduceOp, initValue, expr => expr);
+            // return ReduceCore(op, reduceOp, initValue, expr => expr);
+            var input = GetInputExpr(op, 0);
+            return reduceOp switch
+            {
+                var x when x == ReduceOp.Max && input.CheckedDataType == DataTypes.Float32 => ReduceCore(op, reduceOp, float.MaxValue, expr => expr),
```
Comment (xhuohai) [inline src/Nncase.Importer/Onnx/Reduce.cs:20]: reduceMax's initValue should be xx.MinValue

File: src/Nncase.Importer/Onnx/Reduce.cs [hunk 2]
```diff
@@ -11,12 +11,23 @@ namespace Nncase.Importer
...
+                var x when x == ReduceOp.Max && input.CheckedDataType == DataTypes.Float16 => ReduceCore(op, reduceOp, Half.MaxValue, expr => expr),
+                var x when x == ReduceOp.Max && input.CheckedDataType == DataTypes.Float64 => ReduceCore(op, reduceOp, long.MaxValue, expr => expr),
+                var x when x == ReduceOp.Min && input.CheckedDataType == DataTypes.Float32 => ReduceCore(op, reduceOp, float.MinValue, expr => expr),
+                var x when x == ReduceOp.Min && input.CheckedDataType == DataTypes.Float16 => ReduceCore(op, reduceOp, Half.MinValue, expr => expr),
```
Comment (xhuohai) [inline src/Nncase.Importer/Onnx/Reduce.cs:24]: reduceMin's initValue should be xx.MaxValue

**Review by xhuohai — CHANGES_REQUESTED** [review]

File: src/Nncase.Importer/Onnx/Reduce.cs (hunk 1 above)
Comment (HeJunchao100813) [reply src/Nncase.Importer/Onnx/Reduce.cs]: done

File: src/Nncase.Importer/Onnx/Reduce.cs (hunk 2 above)
Comment (HeJunchao100813) [reply src/Nncase.Importer/Onnx/Reduce.cs]: done

File: src/Nncase.Importer/Onnx/Reduce.cs [hunk 3]
```diff
@@ -51,7 +51,7 @@ private Expr ReduceCore(in NodeProto op, ReduceOp reduceOp, float initValue, Fun
                 var x when x == ReduceOp.Max && input.CheckedDataType == DataTypes.Int32 => F.Tensors.Reduce(reduceOp, f(input), axis, int.MinValue, keepDims),
                 var x when x == ReduceOp.Min && input.CheckedDataType == DataTypes.Int64 => F.Tensors.Reduce(reduceOp, f(input), axis, long.MaxValue, keepDims),
                 var x when x == ReduceOp.Min && input.CheckedDataType == DataTypes.Int32 => F.Tensors.Reduce(reduceOp, f(input), axis, int.MaxValue, keepDims),
-                _ => F.Tensors.Reduce(reduceOp, f(input), axis, F.Tensors.Cast(initValue, input.CheckedDataType), keepDims),
+                _ => F.Tensors.Reduce(reduceOp, f(input), axis, F.Tensors.Cast(initValue, input.CheckedDataType, CastMode.CheckOverflow), keepDims),
```
Comment (xhuohai) [inline src/Nncase.Importer/Onnx/Reduce.cs:54]: Should add bfloat16 and float16 reduce cases in pytest.

**Review by xhuohai — CHANGES_REQUESTED** [review]

File: src/Nncase.Importer/Onnx/Reduce.cs (hunk 3 above)
Comment (HeJunchao100813) [reply src/Nncase.Importer/Onnx/Reduce.cs]: done

**Review by xhuohai — APPROVED** [review]

**Pull request merged**